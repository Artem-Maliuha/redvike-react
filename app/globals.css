@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --primary: #1A1A1E;
  --background-rgb: 255, 255, 255;
}

body {
  color: rgb(var(--primary));
  background: rgb(var(--background-rgb));
}
