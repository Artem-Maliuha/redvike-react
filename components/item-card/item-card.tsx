import Image from 'next/image'
import Link from 'next/link'

interface ItemCardProps {
  name: string
  image: string
  link: string
}
export const ItemCard = (props: ItemCardProps) => {
  const { name, image, link } = props
  return (
    <Link
      href={link}
      className="flex w-[272px] flex-col items-center gap-[12px] border-2 border-black p-[16px]"
    >
      <Image width={100} height={100} src={image} alt={name} />
      <span className="rounded-[4px] bg-[#CFCFCF] px-[4px] text-base font-bold capitalize">
        {name}
      </span>
    </Link>
  )
}
