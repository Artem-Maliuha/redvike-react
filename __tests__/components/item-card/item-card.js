import renderer from 'react-test-renderer'
import { describe, it, expect } from '@jest/globals'
import { ItemCard } from '../../../components/item-card'

describe('<ItemCard />', () => {
  it('snapshot', () => {
    const component = renderer.create(
      <ItemCard name="Name" image="/some.jpg" link="/" />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
