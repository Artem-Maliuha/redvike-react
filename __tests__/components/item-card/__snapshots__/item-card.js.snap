// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemCard /> snapshot 1`] = `
<a
  className="flex w-[272px] flex-col items-center gap-[12px] border-2 border-black p-[16px]"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <img
    alt="Name"
    data-nimg="1"
    decoding="async"
    height={100}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fsome.jpg&w=256&q=75"
    srcSet="/_next/image?url=%2Fsome.jpg&w=128&q=75 1x, /_next/image?url=%2Fsome.jpg&w=256&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={100}
  />
  <span
    className="rounded-[4px] bg-[#CFCFCF] px-[4px] text-base font-bold capitalize"
  >
    Name
  </span>
</a>
`;
